packages/core/content-releases/admin/src/components/AddToReleaseAction.tsx

```diff
@@ -0,0 +1,299 @@
+        defaultMessage:
+          'No available releases. Open the list of releases and create a new one from there.',
+      })}
+      action={
+        <LinkButton
+          to={{
+            pathname: '/plugins/content-releases',
+          }}
+          tag={ReactRouterLink}
+          variant="secondary"
+        >
+          {formatMessage({
+            id: 'content-releases.content-manager-edit-view.add-to-release.redirect-button',
+            defaultMessage: 'Open the list of releases',
+          })}
+        </LinkButton>
+      }
+      shadow="none"
+    />
+  );
+};
+
+const AddActionToReleaseModal = ({
+  contentTypeUid,
+  documentId,
+  onInputChange,
+  values,
+}: AddActionToReleaseModalProps) => {
+  const { formatMessage } = useIntl();
+  const [{ query }] = useQueryParams<{ plugins?: { i18n?: { locale?: string } } }>();
+  const locale = query.plugins?.i18n?.locale;
+
+  // Get all 'pending' releases that do not have the entry attached
+  const response = useGetReleasesForEntryQuery({
+    contentTypeUid,
+    documentId,
+    hasEntryAttached: false,
+    locale,
+  });
+
+  const releases = response.data?.data;
+
+  if (releases?.length === 0) {
+    <NoReleases />;
+  }
+
+  return (
+    <Flex direction="column" alignItems="stretch" gap={2}>
+      <Box paddingBottom={6}>
+        <Field.Root required>
+          <Field.Label>
+            {formatMessage({
+              id: 'content-releases.content-manager-edit-view.add-to-release.select-label',
+              defaultMessage: 'Select a release',
+            })}
+          </Field.Label>
+          <SingleSelect
+            required
+            placeholder={formatMessage({
+              id: 'content-releases.content-manager-edit-view.add-to-release.select-placeholder',
+              defaultMessage: 'Select',
+            })}
+            name="releaseId"
+            onChange={(value) => onInputChange('releaseId', value)}
+            value={values.releaseId}
+          >
+            {releases?.map((release: any) => (
+              <SingleSelectOption key={release.id} value={release.id}>
+                {release.name}
+              </SingleSelectOption>
+            ))}
+          </SingleSelect>
+        </Field.Root>
+      </Box>
+      <Field.Label>
+        {formatMessage({
+          id: 'content-releases.content-manager-edit-view.add-to-release.action-type-label',
+          defaultMessage: 'What do you want to do with this entry?',
+        })}
+      </Field.Label>
+      <ReleaseActionOptions
+        selected={values.type}
+        handleChange={(e) => onInputChange('type', e.target.value)}
+        name="type"
+      />
+    </Flex>
+  );
+};
+
+/* -------------------------------------------------------------------------------------------------
+ * AddToReleaseAction
+ * -----------------------------------------------------------------------------------------------*/
+
+const AddToReleaseAction: DocumentActionComponent = ({
```

Comment:
Would it make more sense to call this `ReleaseActionForm` or with the other suggestion `ReleaseActionModalForm`?